### ISSUE TITLE
Upgrading 1.7.1 to 1.7.2 with RPM causes postun script failed.

### ISSUE THREAD

**Issue by t2y** [elastic/elasticsearch#13565]
Upgrading 1.7.1 to 1.7.2 with RPM causes postun script failed.

```sh
$ sudo rpm -Uvh elasticsearch-1.7.2.noarch.rpm 
warning: elasticsearch-1.7.2.noarch.rpm: Header V4 RSA/SHA1 Signature, key ID d88e42b4: NOKEY
Preparing...                          ################################# [100%]
Updating / installing...
   1:elasticsearch-1.7.2-1            ################################# [ 50%]
Cleaning up / removing...
   2:elasticsearch-1.7.1-1            ################################# [100%]
post remove script called with unknown argument `1'
warning: %postun(elasticsearch-1.7.1-1.noarch) scriptlet failed, exit status 1
```

My environment is here.

```sh
$ lsb_release -a
LSB Version:	:base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Distributor ID:	AmazonAMI
Description:	Amazon Linux AMI release 2015.03
Release:	2015.03
Codename:	n/a
$ rpm --version
RPM version 4.11.2
```

**Comment by t2y** [comment]
elasticsearch-1.7.1.noarch.rpm has a bug about handling the status variable in postun script. But it seems elasticsearch-1.7.2.noarch.rpm has already fixed.

``` #!sh
$ rpm -qp --scripts elasticsearch-1.7.1.noarch.rpm
    ...
    # RedHat ####################################################
    0)
        REMOVE_DIRS=true
        REMOVE_SERVICE=true
        REMOVE_USER_AND_GROUP=true
    ;;
    2)
        # If $1=1 this is an upgrade
        IS_UPGRADE=true
    ;;

    *)
        echo "post remove script called with unknown argument \`$1'" >&2
        exit 1
    ;;
```

**Comment by t2y** [comment]
```sh
$ diff -u <(rpm -qp --scripts elasticsearch-1.7.1.noarch.rpm) <(rpm -qp --scripts elasticsearch-1.7.2.noarch.rpm)
warning: warning: elasticsearch-1.7.1.noarch.rpm: Header V4 RSA/SHA1 Signature, key ID d88e42b4: NOKEY
elasticsearch-1.7.2.noarch.rpm: Header V4 RSA/SHA1 Signature, key ID d88e42b4: NOKEY
--- /dev/fd/63	2015-09-14 23:54:04.798729340 +0000
+++ /dev/fd/62	2015-09-14 23:54:04.802729248 +0000
@@ -81,7 +81,7 @@
     ;;
 esac
 
-# Built for elasticsearch-1.7.1 (rpm)
+# Built for elasticsearch-1.7.2 (rpm)
 postinstall scriptlet (using /bin/sh):
 
 
@@ -183,7 +183,7 @@
     echo " OK"
 fi
 
-# Built for elasticsearch-1.7.1 (rpm)
+# Built for elasticsearch-1.7.2 (rpm)
 preuninstall scriptlet (using /bin/sh):
 
 
@@ -253,7 +253,7 @@
     echo " OK"
 fi
 
-# Built for elasticsearch-1.7.1 (rpm)
+# Built for elasticsearch-1.7.2 (rpm)
 postuninstall scriptlet (using /bin/sh):
 
 
@@ -265,8 +265,8 @@
 #       $1=purge     : indicates an upgrade
 #
 #   On RedHat,
-#       $1=1         : indicates an new install
-#       $1=2         : indicates an upgrade
+#       $1=0         : indicates a removal
+#       $1=1         : indicates an upgrade
 
 
 
@@ -296,7 +296,7 @@
         REMOVE_SERVICE=true
         REMOVE_USER_AND_GROUP=true
     ;;
-    2)
+    1)
         # If $1=1 this is an upgrade
         IS_UPGRADE=true
     ;;
@@ -377,4 +377,4 @@
     fi
 fi
 
-# Built for elasticsearch-1.7.1 (rpm)
+# Built for elasticsearch-1.7.2 (rpm)
```

**Comment by clintongormley** [comment]
Duplicate of #12606. Fixed by #12298
